src/libs/styled/useColorScheme.native.test.ts

```diff
@@ -1,80 +1,89 @@
+  afterEach(() => {
+    jest.clearAllMocks()
   })
 
-  it('should initialize with system colorScheme when is available', () => {
-    mockCurrentColorScheme.mockReturnValueOnce('dark')
-    colorSchemeActions.init()
-    const { result } = renderHook(() => useColorScheme())
+  describe('default', () => {
+    it('should return light when feature flag is disabled', () => {
+      setFeatureFlags([])
+      const { result } = renderHook(() => useColorScheme())
+
+      expect(result.current).toBe(ColorSchemeEnum.LIGHT)
+    })
+
+    it('should return light when feature flag is enable', () => {
+      const { result } = renderHook(() => useColorScheme())
 
-    expect(result.current).toBe('dark')
+      expect(result.current).toBe(ColorSchemeEnum.LIGHT)
+    })
   })
 
-  it.each([null, undefined])(
-    'should default to light theme if no color scheme is detected (%s)',
-    (value) => {
-      mockCurrentColorScheme.mockReturnValueOnce(value)
-      colorSchemeActions.init()
+  describe('user choice', () => {
+    it('should return dark when storedScheme=dark', () => {
+      colorSchemeActions.setColorScheme({ colorScheme: ColorSchemeEnum.DARK })
       const { result } = renderHook(() => useColorScheme())
 
-      expect(result.current).toBe('light')
-    }
-  )
+      expect(result.current).toBe(ColorSchemeEnum.DARK)
+    })
+
+    it('should return light when storedScheme=light', () => {
+      colorSchemeActions.setColorScheme({ colorScheme: ColorSchemeEnum.LIGHT })
+      const { result } = renderHook(() => useColorScheme())
 
-  it('should react to system color scheme change', async () => {
-    colorSchemeActions.init()
-    const { result } = renderHook(() => useColorScheme())
+      expect(result.current).toBe(ColorSchemeEnum.LIGHT)
+    })
 
-    expect(result.current).toBe('light')
+    it('should return light when storedScheme=system is light', () => {
+      jest.spyOn(ReactNative, 'useColorScheme').mockReturnValueOnce(ColorSchemeEnum.LIGHT)
+      colorSchemeActions.setColorScheme({ colorScheme: ColorSchemeEnum.SYSTEM })
+      const { result } = renderHook(() => useColorScheme())
 
-    await act(async () => {
-      Appearance.setColorScheme('dark')
+      expect(result.current).toBe(ColorSchemeEnum.LIGHT)
     })
 
-    expect(result.current).toBe('dark')
-  })
+    it('should return dark when storedScheme=system is dark and colorScheme=dark', () => {
+      jest.spyOn(ReactNative, 'useColorScheme').mockReturnValueOnce(ColorSchemeEnum.DARK)
+      colorSchemeActions.setColorScheme({ colorScheme: ColorSchemeEnum.SYSTEM })
+      const { result } = renderHook(() => useColorScheme())
 
-  it.each([null, undefined])(
-    'should default to light theme when system color scheme becomes unavailable (%s)',
-    async (value) => {
-      colorSchemeActions.init()
+      expect(result.current).toBe(ColorSchemeEnum.DARK)
+    })
+
+    it('should return dark when storedScheme=system is dark but feature flag is disabled', () => {
```

Comment:
le nom du test est faux par rapport à ce qui est réellement testé

```suggestion
    it('should return light when storedScheme=system is dark but feature flag is disabled', () => {
```